client/lib/posts/post-list-store.js

```diff
@@ -28,344 +33,350 @@ var _defaultQuery = {
-		sort( _activeList.postIds );
+		if ( newPostIds.length ) {
+			_activeList.postIds = _activeList.postIds.concat( newPostIds );
+			sort( _activeList.postIds );
+		}
 	}
 
-	PostListStore.emit( 'change' );
-}
+	return new class extends EventEmitter {
+		constructor() {
+			super();
+			this.id = id;
+			this.dispatchToken = Dispatcher.register( this.handlePayload.bind( this ) );
+		}
 
-PostListStore = {
+		get() {
+			return _activeList;
+		}
 
-	get: function() {
-		return _activeList;
-	},
+		getID() {
+			return _activeList.id;
+		}
 
-	getID: function() {
-		return _activeList.id;
-	},
+		getSiteID() {
+			return _activeList.query.siteID;
+		}
 
-	getSiteID: function() {
-		return _activeList.query.siteID;
-	},
+		// Get list of posts from current object
+		getAll() {
+			return _activeList.postIds.map( function( globalID ) {
+				return PostsStore.get( globalID );
+			} );
+		}
 
-	/**
-	* Get list of posts from current object
-	*/
-	getAll: function() {
-		return _activeList.postIds.map( function( globalID ) {
-			return PostsStore.get( globalID );
-		} );
-	},
+		getTree() {
+			const sortedPosts = [];
 
-	getTree: function() {
-		var sortedPosts = [];
+			// clone objects to prevent mutating store data, set parent to number
+			_activeList.postIds.forEach( function( globalID ) {
+				let post = clone( PostsStore.get( globalID ) );
+				post.parent = post.parent ? post.parent.ID : 0;
+				sortedPosts.push( post );
+			} );
 
-		// clone objects to prevent mutating store data, set parent to number
-		_activeList.postIds.forEach( function( globalID ) {
-			var post = clone( PostsStore.get( globalID ) );
-			post.parent = post.parent ? post.parent.ID : 0;
-			sortedPosts.push( post );
-		} );
+			return treeConvert( 'ID' ).treeify( sortedPosts );
+		}
 
-		return treeConvert.treeify( sortedPosts );
-	},
+		getPost( globalID ) {
+			if ( _activeList.postIds.indexOf( globalID ) > -1 ) {
+				return PostsStore.get( globalID );
+			}
+		}
 
-	getPost: function( globalID ) {
-		if ( _activeList.postIds.indexOf( globalID ) > -1 ) {
-			return PostsStore.get( globalID );
+		getPage() {
+			return _activeList.page;
 		}
-	},
-
-	getPage: function() {
-		return _activeList.page;
-	},
-
-	isLastPage: function() {
-		return _activeList.isLastPage;
-	},
-
-	isFetchingNextPage: function() {
-		return _activeList.isFetchingNextPage;
-	},
-
-	getNextPageParams: function() {
-		var params = {},
-			query = _activeList.query;
-
-		params.status = query.status;
-		params.order_by = query.orderBy;
-		params.order = query.order;
-		params.author = query.author;
-		params.number = query.perPage;
-		params.type = query.type;
-		params.page_handle = _activeList.nextPageHandle;
-		params.exclude_tree = query.exclude_tree;
-		params.number = query.number;
-		params.before = query.before;
-		params.after = query.after;
-
-		if ( query.search ) {
-			params.search = query.search;
+
+		off( event, method ) {
+			this.removeListener( event, method );
 		}
```

Comment:
I think we should get rid of `off` and use `removeListener` instead.  Probably outside the scope of this PR though.